### PR TITLE
Issue 10262: Don't accept BCC posts from non followers

### LINKED ISSUE #10262: Receiving unwanted posts from Hubzilla followers
This is the second time in a couple days that I've received posts from Hubzilla addressed to me in Bcc. In both cases the account is following me but I am not following them. First time felt like a bug, the second time feels like spam.

I feel like we should check whether the target local user is following a remote account before delivering any post. Further than that, I suggest we drop the post entirely so that it can't show in the community pages either, since it looks like an easy way to spread spam: follow a single account on all the Friendica nodes you can think of, post something with all the accounts in Bcc, even if the target users don't receive it in their timeline, it still is recorded by the Friendica node and displayed in the global community page.

### PR DESCRIPTION
**Pull request by annando**
Fixes #10262

### CODE REVIEW

File: src/Protocol/ActivityPub/Processor.php [hunk 1]
```diff
@@ -602,6 +602,12 @@ public static function postItem(array $activity, array $item)
 				continue;
 			}
 
+			if (!$item['isForum'] && ($receiver != 0) && ($item['gravity'] == GRAVITY_PARENT) &&
+				($item['post-reason'] == Item::PR_BCC) && !Contact::isSharingByURL($activity['author'], $receiver)) {
+				Logger::info('Top level post via BCC from a non follower, ignoring', ['uid' => $receiver, 'contact' => $item['contact-id']]);
```
Comment (MrPetovan) [inline src/Protocol/ActivityPub/Processor.php]: Do you mean from an account that isn't followed?

Comment (annando) [reply src/Protocol/ActivityPub/Processor.php]: I'm always confused with follower, following, sharing and so on :-) So: How should I name it?

Comment (tobiasd) [reply src/Protocol/ActivityPub/Processor.php]: in my mind, it is always a Stalker... no you should not call it that way.

They follow you, but you don't follow them. I'd say

> Top level post via BCC from an account that isn't followed, ignoring

Comment (MrPetovan) [reply src/Protocol/ActivityPub/Processor.php]: I'm with Tobias.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**